rust/otel-arrow-rust/src/schema.rs

```diff
@@ -144,3 +145,40 @@ pub fn is_id_plain_encoded(record_batch: &RecordBatch) -> bool {
     );
     encoding == Some(consts::metadata::encodings::PLAIN)
 }
+
+/// Checks the Arrow schema field metadata to determine if the "parent_id" field in this record
+/// batch is plain encoded.
+///
+/// It is assumed that if the Parent ID column is plain encoded, the schema metadata will have the
+/// encoding identified explicitly. If the encoding metadata is absent, then the batch may have
+/// been produced by the golang exporter, which adds no metadata and generally uses the transport
+/// optimized/quasi-delta encoding for the Parent ID column by default.
+#[must_use]
+pub fn is_parent_id_plain_encoded(record_batch: &RecordBatch) -> bool {
+    let schema = record_batch.schema_ref();
+    let encoding = get_field_metadata(
+        schema.as_ref(),
+        consts::PARENT_ID,
+        consts::metadata::COLUMN_ENCODING,
+    );
+    encoding == Some(consts::metadata::encodings::PLAIN)
+}
+
+/// Additional helper methods for Arrow Field
+pub trait FieldExt {
+    /// Sets the encoding column metadata key to "plain".
```

Comment:
It looks like `Field::with_metadata` method would replace and not append to any existing metadata on the `Field`. Maybe that's okay since we know the call sites for this method are not made on fields which have any existing metadata, but it might be good to call it out in the documentation here or offer another method which appends the metadata?

Reply:
Great catch @utpilla! Fixed this change to avoid confusion. The intention here was only to update the "encoding" key, not replace the entire metadata.